### ISSUE TITLE
Extracting/Migrating the Credential Provider

### ISSUE THREAD

**Issue by andrewsykim** [kubernetes/cloud-provider#13]
As part of the cloud provider extraction/migration, we should start to look into how the credential provider is going to be extracted so that the kubelet does not rely on cloud SDKs for image pulling credentials. Also to support future credential providers without adding it into the main tree. 

Need to work with SIG Auth and propose a KEP to extract/migrate credential providers to move out-of-tree. 

related: https://github.com/kubernetes/kubernetes/issues/68810 

cc @justinsb @mcrute

**Comment by mcrute** [comment]
Also related kubernetes/kubernetes#70675

**Comment by andrewsykim** [comment]
@mcrute and I are doodling out some ideas for this, hoping we can have an alpha feature-gated version of this in for v1.15. 

cc @liggitt @smarterclayton

**Comment by feiskyer** [comment]
Would secret a possible solution for this?

**Comment by andrewsykim** [comment]
> Would secret a possible solution for this?

Yes, this is one of the possible solutions we're discussing. KEP is work in progress :)

**Comment by andrewsykim** [comment]
For KEP + alpha implementation

/milestone v1.15 
/priority critical-urgent

cc @dchen1107 @derekwaynecarr

**Comment by andrewsykim** [comment]
p0: KEP + alpha implementation

**Comment by mcrute** [comment]
[Rough draft](https://docs.google.com/document/d/1QIU2NxwsMbDybE4RmMsMiZqANyxvUfyQyMiW8j_1cf4/edit) of the KEP, mostly just the problem statement at this point.

**Comment by feiskyer** [comment]
> For v1.15, p0: KEP + alpha implementation

@andrewsykim What's your proposal for alpha implementation?
